xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -260,18 +261,29 @@ void CLinuxRendererGLES::LoadPlane(CYuvPlane& plane, int type,
   glBindTexture(m_textureTarget, plane.id);
 
   // OpenGL ES does not support strided texture input.
+  GLint pixelStore = -1;
   if (stride != static_cast<int>(width * bps))
   {
-    unsigned char* src = static_cast<unsigned char*>(data);
-    for (unsigned int y = 0; y < height; ++y, src += stride)
+    unsigned int verMajor, verMinor;
+    m_renderSystem->GetRenderVersion(verMajor, verMinor);
+
+    if (verMajor < 3)
     {
-      glTexSubImage2D(m_textureTarget, 0, 0, y, width, 1, type, GL_UNSIGNED_BYTE, src);
+      unsigned char* src (static_cast<unsigned char*>(data)), *dst(m_planeBuffer);
+      for (unsigned int y = 0; y < height; ++y, src += stride, dst += width)
+        memcpy(dst, src, width);
+      pixelData = m_planeBuffer;
```

Comment:
this is only correct for bps == 1. As the render seems to not support more anyways I'd return early if bps != 1